src/ProjectService.Business/Commands/EditTaskCommand.cs

```diff
@@ -0,0 +1,121 @@
+        private readonly IRequestClient<IGetDepartmentRequest> _requestClient;
+        
+        private IGetDepartmentResponse GetDepartment(Guid userId, List<string> errors)
+        {
+            string errorMessage = "Cannot edit task. Please try again later.";
+
+            try
+            {
+                var response = _requestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(
+                    IGetDepartmentRequest.CreateObj(userId, null)).Result;
+
+                if (response.Message.IsSuccess)
+                {
+                    return response.Message.Body;
+                }
+
+                _logger.LogWarning($"Can not find department with this id '{userId}': " +
+                                   $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
+            }
+            catch (Exception exc)
+            {
+                _logger.LogError(exc, errorMessage);
+
+                errors.Add(errorMessage);
+            }
+
+            return null;
+        }
+
+        public EditTaskCommand(
+            ITaskRepository taskRepository,
+            IProjectRepository projectRepository,
+            IEditTaskValidator validator,
+            IAccessValidator accessValidator,
+            IHttpContextAccessor httpContextAccessor,
+            IPatchDbTaskMapper mapper,
+            ILogger<EditTaskCommand> logger,
+            IRequestClient<IGetDepartmentRequest> requestClient)
+        {
+            _taskRepository = taskRepository;
+            _projectRepository = projectRepository;
+            _validator = validator;
+            _accessValidator = accessValidator;
+            _httpContext = httpContextAccessor.HttpContext;
+            _mapper = mapper;
+            _logger = logger;
+            _requestClient = requestClient;
+        }
+
+        public OperationResultResponse<bool> Execute(Guid taskId, JsonPatchDocument<EditTaskRequest> patch)
+        {
+            var errors = new List<string>();
+
+            DbTask task = _taskRepository.Get(taskId);
+
+            DbProject project = _projectRepository.GetProject(task.ProjectId);
+
+            Guid requestUserId = _httpContext.GetUserId();
+
+            IGetDepartmentResponse department = GetDepartment(requestUserId, errors);
+
+            bool isAdmin = _accessValidator.IsAdmin();
+            
+            bool isProjectParticipant = project?.Users.FirstOrDefault(x =>
+                x.UserId == requestUserId) != null;
+            
+            bool isDepartmentDirector = department?.Id == project?.DepartmentId;
+            
+            if (!isAdmin && !isProjectParticipant && !isDepartmentDirector)
+            {
+                throw new ForbiddenException("Not enough rights.");
+            }
+
+            _validator.ValidateAndThrowCustom(patch);
+
+            var dbTaskPatch = _mapper.Map(patch);
+            _taskRepository.Edit(taskId, dbTaskPatch);
```

Comment:
You can throw into _taskRepository.Edit(DbTask task(from 88str.), dbTaskPatch) without second search the project by ID in DB. It will work.